src/main/java/com/blackhearth/blockchain/node/BlockChainNode.java

```diff
@@ -14,18 +17,19 @@
 @Component
 @RequiredArgsConstructor
 public class BlockChainNode {
-
     private final Random random;
 
     private boolean isNodeRunning = false;
     @Getter
     private ServerSocket socket;
 
+    @Autowired
```

Comment:
Zamiast @Autowired użyj słowa kluczowego final (@RequiredArgsConstructor) tworzy konstruktor ze wszystkich pól które są final, a to następnie injectuje wszystkie beany.